lib/cli/cliCompact.js

```diff
@@ -0,0 +1,181 @@
+                    console.log('Compact group:          ' + (obj.common.compactGroup || 0));
+                    return void callback();
+                } else {
+                    messages.error.invalidInstance(instance);
+                    return void callback(24);
+                }
+            });
+        });
+    }
+
+    setCompactModeInstance(instance, targetState, groupId) {
+        const { callback, dbConnect } = this.options;
+
+        dbConnect((objects, _states, isOffline, _objectDbType, config) => {
+            if (!this.config.system || !this.config.system.compact) {
+                console.log('Compact mode is disabled on this host. This setting will have no effect!');
+                console.log();
+            }
+            objects.getObject('system.adapter.' + instance, (err, obj) => {
+                if (!err && obj) {
+                    console.log('Compact mode supported: ' + !!obj.common.compact);
+                    let newRunAsCompactMode;
+                    if (targetState !== undefined && targetState !== !!obj.common.runAsCompactMode) {
+                        newRunAsCompactMode = targetState;
+                    }
+                    let newCompactGroup;
+                    if (obj.common.compactGroup === undefined) {
+                        obj.common.compactGroup = obj.common.compactGroup || 0;
+                        newCompactGroup = obj.common.compactGroup;
+                    }
+                    if (groupId !== undefined) {
+                        groupId = parseInt(groupId, 10);
+                        if (groupId !== obj.common.compactGroup) {
+                            newCompactGroup = groupId;
+                        }
+                    }
+                    console.log('Compact mode enabled:   ' + !!obj.common.runAsCompactMode + (newRunAsCompactMode !== undefined ? ' --> ' + newRunAsCompactMode : ''));
+                    console.log('Compact group:          ' + obj.common.compactGroup + (newCompactGroup !== undefined ? ' --> ' + newCompactGroup : ''));
+                    if (newRunAsCompactMode !== undefined || newCompactGroup !== undefined) {
+                        if (newCompactGroup !== undefined) {
+                            obj.common.compactGroup = newCompactGroup;
+                        }
+                        if (newRunAsCompactMode !== undefined) {
+                            obj.common.runAsCompactMode = newRunAsCompactMode;
+                        }
+
+                        obj.from = 'system.host.' + tools.getHostName() + '.cli';
+                        obj.ts = new Date().getTime();
+                        objects.setObject('system.adapter.' + instance, obj, (err) => {
+                            if (err) console.log(err);
+                            console.log('Instance settings for "' + instance + '" are changed.');
+                            return void callback();
```

Comment:
The `void` is not really necessary here since it is inside another callback. But doesn't hurt.